src/modules/mavlink/mavlink_main.cpp

```diff
@@ -1310,6 +1310,14 @@ void Mavlink::send_autopilot_capabilites()
 		board_get_uuid32(uid);
 		msg.uid = (((uint64_t)uid[PX4_CPU_UUID_WORD32_UNIQUE_M]) << 32) | uid[PX4_CPU_UUID_WORD32_UNIQUE_H];
 
+#ifdef CONFIG_ARCH_BOARD_SITL
```

Comment:
hmm. We should avoid BOARD based ifdefs. The board support function board_get_uuid32 would be the place to add the place to create a real UUID. But I can see if the uniqueness is based on mavlink_system.sysid that may pose an issue.

Reply:
Right, I'm open for suggestions how to do that.